docs/contributing.rst

```diff
@@ -165,6 +165,58 @@ subset of the test suite. In order to run a specific test run:
 
 The entire test suite is automatically run for each pull request.
 
+Performance measurement
+-----------------------
+
+To track the performance of various functionality of the modin project, we use the asv_ tool.
+
+Various scenarios in which it may be needed with appropriate command:
+
+* It is necessary to check the impact of the new patch on the performance of a certain set of operations:
+
+.. code-block:: bash
+
+  asv continuous -f 1.05 src/master HEAD -b TimeGroupBy --launch-method=spawn
+
+* It is necessary to check the healthy of benchmarks after making changes or writing new ones:
+
+.. code-block:: bash
+
+  asv run --quick --show-stderr --python=same --launch-method=spawn
+
+* You just need to run the entire test suite to get the current time numbers:
+
+.. code-block:: bash
+
+  asv run --launch-method=spawn
+
+* It is necessary to check the range of commits for performance degradation:
+
+.. code-block:: bash
+
+ asv run [start_hash]..[end_hash] --launch-method=spawn
+ asv publish
+ asv preview
+
+For more consistent results, you may need to use the following parameters:
+
+* ``-a sample_time=1``
+* ``-a warmup_time=1``
+* ``-a processes=4``
+
```

Comment:
We should also add some information about `-a number=N` parameter, since some of our benchmarks may take a time that doesn't fit into sample time (especially when `MODIN_TEST_DATASET_SIZE=Big`), so since `sample_time / run_time <= 1` there will be only a one repeat of a benchmark (correct me if I'm wrong). So to make benchmark results more consistent, we should explicitly set `-a number=N`

Reply:
Minimum number of benchmark repeats is `2` according to [min_run_count](https://asv.readthedocs.io/en/stable/benchmarks.html?highlight=sample_time#timing-benchmarks) parameter.

On a machine specially prepared for benchmarks, if the time of one run is more than 1 second, then fairly stable results are obtained. (for example, you can increase `sample_time`)

Reply:
okay, got it